### PR TITLE
First efforts towards an improved improved readme

### PR DESCRIPTION
**Pull request by Samuwhale**
Note, at one point the readme refers to tutorials in the documentation. For now this hyperlink simply directs to the index page, as there is no tutorial page in the docs yet. This will be updated ofcourse later once this page is present in the docs.

### CODE REVIEW

File: README.md
Comment (Samuwhale) [reply README.md:8]: Yep, agreed

Comment (Samuwhale) [reply README.md:8]: You you're right, the idea was to provide an incentive over why you'd want to specifically use metasynth over potential other solutions, but without specifics it reads vague and just makes the sentence more verbose. I'll keep an overview/comparison on my readthedocs to-do list :)

File: README.md
Comment (Samuwhale) [reply README.md:20]: yeah I already was thinking it's a bit hacky, but for the user it probably makes no difference hence why I went for this quick approach to quickly be able to test images. 

I'll add the images directly to the repo (in the readthedocs folder, so they can be used there later too), so that they can be linked :)

File: README.md
Comment (Samuwhale) [reply README.md:261]: Yeah, I was planning on tucking it away in the readthedocs (alongside some other things). Reason I had not yet done so was that I didn't want to have the information be absent temporarily before it was added to the docs. But I'll remove it as I do think it makes this paragraph more confusing than that it helps clarify things.

File: README.md
Comment (Samuwhale) [reply README.md:263]: This section is based on the ONS description 

> Preserve the format and record-level plausibility as detailed previously and replicate marginal (univariate) distributions where possible.
> 
> Constructed based on the real dataset, values are generated based on observed distributions (with added fuzziness and smoothing) but no attempt made to preserve relationships.
> 
> Missing value codes and their frequency is to be preserved.
> 
> Disclosure control evaluation is necessary case by case, special care to be taken with names and so on.
> 
> To be used for extended code testing, minimal analytical value, non-negligible disclosure risk.

But I think it can be removed as it might just make the user more confused, it's something that can potentially be added and elaborated on later in the readthedocs.

File: README.md
Comment (Samuwhale) [reply README.md:266]: None at all, just a sloppy oversight on my part.

File: README.md
Comment (Samuwhale) [reply README.md:303]: I was doubting it, reason I initially did not was that this way you make it more tangible to the user what the dataset is (as opposed to being the result of a magic function) and where it comes from (as clicking on the link shows the csv). In hindsight, I think the following works too, allowing the user to see what the dataset is while also showcasing the function:

```python
# import libraries
import polars as pl
import metasynth as ms

dataset_csv = ms.demo_file() # This automatically loads the Titanic dataset (as found here https://raw.githubusercontent.com/sodascience/metasynth/main/metasynth/demo/demo_titanic.csv) 

# create dataframe
dtypes = {
    "Sex": pl.Categorical,
    "Embarked": pl.Categorical,
    "Survived": pl.Categorical,
    "Pclass": pl.Categorical,
    "SibSp": pl.Categorical,
    "Parch": pl.Categorical
}

df = pl.read_csv(dataset_csv, dtypes=dtypes)
```

File: README.md
Comment (Samuwhale) [reply README.md:378]: Agreed, for now I think it's good to keep the error warnings in. But, I'll try resolving these issues in the near future so that the warnings can be removed :)

File: README.md
Comment (Samuwhale) [reply README.md:10]: Not sure what happened there, but I'll fix it :)

File: README.md
Comment (Samuwhale) [reply README.md]: I rewrote the introduction to be more concise while also providing more insight on how it improves privacy/reproducibility

**Comment by Samuwhale** [comment]
I've improved the readme based on the comments :)

File: README.md
Comment (qubixes) [reply README.md:155]: Honestly, you might have been right. It's probably easier for researchers to extrapolate to using their own dataset. I'll leave it up to you to decide which solution is the best.

File: README.md
Comment (qubixes) [reply README.md:231]: Yes, it will be fixed one way or the other, so you can just remove it in this PR and assume the warning doesn't get thrown.

**Review by qubixes — APPROVED** [review]
Good work! You have my approval for merging this PR if you remove the text about the deprecation warning (and the warning itself). @vankesteren Can you look through it and merge it if you agree?

**Comment by Samuwhale** [comment]
cool, removed the deprecation warnings. Should be good to go :)

File: README.md
Comment (vankesteren) [inline README.md:24]:
```suggestion
<summary> A simple example of a GMF file: </summary>
```

**Review by vankesteren — APPROVED** [review]
Very nice work! Approved after fixing my super tiny comments :)

**Comment by Samuwhale** [comment]
Should be all good to go!

**Pull request merged**
